language/move-lang/tests/move_check/typing/lambda.move

```diff
@@ -0,0 +1,72 @@
+module 0x8675309::M {
+    use 0x1::XVector;
+
+    public macro foreach<T>(v: &vector<T>, action: |&T|) { // expected to be not implemented
+        let i = 0;
+        while (i < XVector::length(v)) {
+            action(XVector::borrow(v, i));
+            i = i + 1;
+        }
+    }
+
+    public fun correct() {
+        let v = vector[1, 2, 3];
+        let sum = 0;
+        foreach!(&v, |e| sum = sum + *e) // expected to be not implemented
+    }
+
+    public macro wrong_local_call_arg_count<T>(v: &vector<T>, action: |&T|) {
+        let i = 0;
+        while (i < XVector::length(v)) {
+            action(XVector::borrow(v, i), i); // expected to have wrong argument count
+            i = i + 1;
+        }
+    }
+
+    public macro wrong_local_call_arg_type<T>(v: &vector<T>, action: |&T|) {
+        let i = 0;
+        while (i < XVector::length(v)) {
+            action(i); // expected to have wrong argument type
+            i = i + 1;
+        }
+    }
+
+    public macro wrong_local_call_result_type<T>(v: &vector<T>, action: |&T|) {
+        let i = 0;
+        while (i < XVector::length(v)) {
+            i = i + action(XVector::borrow(v, i)); // expected to have wrong result type
+        }
+    }
+
```

Comment:
Other ideas for tests:
- a macro with a normal return type type-checks
- a macro's return type cannot be a lambda
- a normal function's return type cannot be a lambda
could be useful

Reply:
Added those tests.